debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ReplicaSet.java

```diff
@@ -5,74 +5,25 @@
  */
 package io.debezium.connector.mongodb;
 
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
 import java.util.Objects;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-
-import com.mongodb.ServerAddress;
 
 import io.debezium.annotation.Immutable;
 
 @Immutable
 public final class ReplicaSet implements Comparable<ReplicaSet> {
 
-    /**
-     * Regular expression that extracts the hosts for the replica sets. The raw expression is
-     * {@code ((([^=]+)[=])?(([^/]+)\/))?(.+)}.
-     */
-    private static final Pattern HOST_PATTERN = Pattern.compile("((([^=]+)[=])?(([^/]+)\\/))?(.+)");
-
-    /**
-     * Parse the supplied string for the information about the hosts for a replica set. The string is a shard host
-     * specification (e.g., "{@code shard01=replicaSet1/host1:27017,host2:27017}"), replica set hosts (e.g.,
-     * "{@code replicaSet1/host1:27017,host2:27017}"), or standalone host (e.g., "{@code host1:27017}" or
-     * "{@code 1.2.3.4:27017}").
-     *
-     * @param hosts the hosts string; may be null
-     * @return the replica set; or {@code null} if the host string could not be parsed
-     */
-    public static ReplicaSet parse(String hosts) {
-        if (hosts != null) {
-            Matcher matcher = HOST_PATTERN.matcher(hosts);
-            if (matcher.matches()) {
-                String shard = matcher.group(3);
-                String replicaSetName = matcher.group(5);
-                String host = matcher.group(6);
-                if (host != null && host.trim().length() != 0) {
-                    return new ReplicaSet(host, replicaSetName, shard);
-                }
-            }
-        }
-        return null;
-    }
-
```

Comment:
Ah, I see you moved this. What was the thinking?

Reply:
To clean up Replica Set to only contain the information which is needed. ReplicaSets are no longer (outside of some testing code) constructed by parsing hosts.

Reply:
As a matter of fact `ReplicaSet` does not event contain list of addresses anymore. It's just a wrapper around name (and possibly shardName). Technically the `ResplicaSet` class could be removed and replace just by a String (the name), however this 1) was less invasive 2) is more extensible if we also want to keep track of what a primary of that replica is 3) It reads better